Site Design Screen Improvements

ovitrif opened the pull request:
## Description
This PR implements the Site Design Screen Improvements (internal ref: `pbArwn-4jE-p2`)

Merged subtasks:
* [Remove unused string resource hpp_subtitle](https://github.com/wordpress-mobile/WordPress-Android/pull/16419)
* [Remove bottom toolbar](https://github.com/wordpress-mobile/WordPress-Android/pull/16415)
* [Remove thumbnail mode selector](https://github.com/wordpress-mobile/WordPress-Android/pull/16414)
* [Remove category filter pills](https://github.com/wordpress-mobile/WordPress-Android/pull/16428)
* [Enlarges design thumbnails](https://github.com/wordpress-mobile/WordPress-Android/pull/16474)
* [Open preview on thumbnail tap](https://github.com/wordpress-mobile/WordPress-Android/pull/16480)
* [Main View - Hide layouts row separators](https://github.com/wordpress-mobile/WordPress-Android/pull/16488)
* [Adds recommended designs section](https://github.com/wordpress-mobile/WordPress-Android/pull/16486)
* [Randomize order of designs for categories](https://github.com/wordpress-mobile/WordPress-Android/pull/16531)
* [Send recommended property with Tracks event on design selection](https://github.com/wordpress-mobile/WordPress-Android/pull/16538)
* [Recommends designs based on chosen vertical](https://github.com/wordpress-mobile/WordPress-Android/pull/16510)
* [Fixes crash occurring when rotating the Site Intent Question screen](https://github.com/wordpress-mobile/WordPress-Android/pull/16546)
* [Updates the UI on vertical change](https://github.com/wordpress-mobile/WordPress-Android/pull/16549)
* [Fix screen titles visibility on tablet for the new & updated screens](https://github.com/wordpress-mobile/WordPress-Android/pull/16550)
* [Align screen titles to left (right for RTL)](https://github.com/wordpress-mobile/WordPress-Android/pull/16556)
* [Add helper text to the bottom of the theme picker](https://github.com/wordpress-mobile/WordPress-Android/pull/16555)
* [Replace Dagger with Hilt for Dependency Injection](https://github.com/wordpress-mobile/WordPress-Android/pull/16575)
* [Sets the loading skeleton to the size of the recommended row](https://github.com/wordpress-mobile/WordPress-Android/pull/16597)
* [Fixes a UI regression by setting the correct height in the Page Layout Picker skeleton view](https://github.com/wordpress-mobile/WordPress-Android/pull/16599)
* [Preload MShots when the site creation activity starts](https://github.com/wordpress-mobile/WordPress-Android/pull/16584)
* [Tweak UI of Site Theme to match design specs](https://github.com/wordpress-mobile/WordPress-Android/pull/16619)
* [Use the custom Glide loader for preloading the thumbnails](https://github.com/wordpress-mobile/WordPress-Android/pull/16624)

## To test
Follow the test plan at `pc8HXX-lA-p2`

## Regression Notes
1. Potential unintended areas of impact
Page Layout Picker

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing automated tests (`ModalLayoutPickerViewModelTest`)

3. What automated tests I added (or what prevented me from doing so)
* Updated `SiteCreationMainVMTest` and `HomePagePickerViewModelTest`
* Added `SiteDesignRecommendationProviderTest`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mkevins committed:
Remove thumbnail mode selector from design picker

mkevins committed:
Rename theme chooser title

mkevins committed:
Remove subtitle text from theme chooser heading

mkevins committed:
Move lambdas outsite of parentheses

This isn't needed for this task, but is useful to satisfy the IDE :)

mkevins committed:
Remove bottom toolbar from theme chooser screen

mkevins committed:
Add TODOS for loose ends from removed UI

Since we are removing some parts of an implementation that shares a base
class with the page picker, there will be some changes needed to adapt
these for the new flow. The intent with these comments is to serve as a
place-holder denoting some things that need to be rewired.

mkevins committed:
Fix lint issue

mkevins committed:
Merge branch 'feature/site-design-revamp--remove-thumbnail-mode-selector' into feature/site-design-revamp--remove-bottom-toolbar

mkevins committed:
Fix lint issue

ovitrif committed:
Remove string resource hpp_subtitle that becomes unused

ovitrif committed:
Hide the category pills recyclerview and skeleton view

ovitrif committed:
Set text alignment of the big title to the left of the screen

ovitrif committed:
Add horizontal padding to the big title to align it with the preview cards

mkevins committed:
Merge pull request #16419 from wordpress-mobile/pr/16414-remove-unused-string-res

[Site design revamp] Remove unused string resource hpp_subtitle

mkevins committed:
Use mobile thumbnails in theme picker screen

mkevins committed:
Reword the test cases to reflect that thumbnails are not changed

mkevins committed:
Merge branch 'feature/site-design-revamp--remove-thumbnail-mode-selector' into feature/site-design-revamp--remove-bottom-toolbar

mkevins committed:
Remove thumbnail preview mode tracking in theme picker

ovitrif committed:
Merge pull request #16415 from wordpress-mobile/feature/site-design-revamp--remove-bottom-toolbar

[Site design revamp] Remove bottom toolbar

ovitrif committed:
Remove unusd resource hpp_button_max_width

ovitrif committed:
Merge pull request #16414 from wordpress-mobile/feature/site-design-revamp--remove-thumbnail-mode-selector

[Site design revamp] Remove thumbnail mode selector

ovitrif committed:
Align collapsed title text to left of Site Theme screen

ovitrif committed:
Merge pull request #16428 from wordpress-mobile/feature/site-design-revamp--remove-category-filter-pills

[Site Design Revamp]  Remove category filter pills

ovitrif committed:
Merge branch 'trunk' into feature/site-design-revamp

ovitrif committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins commented on WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerTracker.kt:
```diff
@@ -3,7 +3,8 @@ package org.wordpress.android.ui.layoutpicker
     fun trackPreviewModeChanged(mode: String)
 
-    fun trackThumbnailModeTapped(mode: String)
+    fun trackThumbnailModeTapped(mode: String) {
```
Perhaps I should have left a comment here, something like:
// NOOP implementation provided to make override optional in subclasses

antonis committed:
Sets the layout picker thumbnail layout dimension dynamically

antonis committed:
Sets the loading skeleton layout dimension dynamically

antonis committed:
Uses a different thumb dimension for tablet devices

antonis committed:
Sets the category row height dynamically

mkevins committed:
Present preview when the theme is tapped

This also refreshes the uiState of the picker, since it is still
partially visible on wider screens.

mkevins committed:
Deselect theme when the preview is dismissed

This also makes loadLayouts non-private, which isn't ideal. It seems
cleaner than adding another set of functions, though.

mkevins committed:
Hide selected overlay (check mark) in theme picker

antonis committed:
Merge pull request #16474 from wordpress-mobile/task/16408-enlarge-thumbnails

[Site Design Revamp] Enlarges design thumbnails

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--open-preview-on-thumbnail-tap

mkevins committed:
Add content description for preview action to theme thumbnails

This also renames the LayoutListItemUiState parameter to be more
meaningful for the way in which it is used.

mkevins committed:
Update theme picker tests

This also includes a commented out test for discussion purposes, which
will be removed in the following commit.

mkevins committed:
Remove unused test in theme picker viewmodel

mkevins committed:
Add comment on tracker default NOOP implementation

This is not part of the PR's underlying task, but seems too small for
its own stand-alone PR.

antonis committed:
Merge pull request #16480 from wordpress-mobile/feature/site-design-revamp--open-preview-on-thumbnail-tap

[Site design revamp] Open preview on thumbnail tap

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Extracts default implementation for the scale property of the ThumbDimensionProvider

antonis committed:
Adds dimension provider for recommended designs

antonis committed:
WIP: Adds recommended designs row

antonis committed:
Fixes lint issue

ovitrif committed:
Add id to divider view between layouts rows

ovitrif committed:
Add showDividers bool to layouts item view holder for showing/hiding the separators

ovitrif committed:
Hide layouts row separator on site design screen

antonis committed:
Adds recommended title

antonis committed:
Merge pull request #16488 from wordpress-mobile/task/16487-hide-separators

[Site Design Revamp] Main View - Hide layouts row separators

antonis committed:
Fixes broken tests

antonis committed:
Workaround to prevent recyclerview from recycling rows of different size

mkevins committed:
Override getItemViewType to re-enable recycling

mkevins committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Merge branch 'feature/site-design-revamp' into task/16393-recommended-designs

antonis committed:
Minor code optimization to avoid force unwrapping

antonis committed:
Extracts (dummy) recommendation logic to a separate function

mkevins committed:
Add bottom helper text to theme picker

This implements an optional footer parameter to layouts recycle view,
useful for working around scroll issues in the screen where it is used.
It introduces a new abstraction `LayoutRowViewHolder` from which
`LayoutItemViewHolder` and `LayoutFooterViewHolder` extend.

The footer is inflated when needed from the passed resource id (the
overriden item count is incremented when not null).

It also adds the footer as a `LayoutCategoryViewType` so that it isn't
mixed with the regular items during view recycling.

antonis committed:
Sets Blogging as the default fallback recommendation

antonis committed:
Mocks resource provider call with one parameter

ovitrif committed:
Merge pull request #16486 from wordpress-mobile/task/16393-recommended-designs

[Site Design Revamp] Adds recommended designs section

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Updates FluxC reference

antonis committed:
Pass selected vertical to the design picker

antonis committed:
Recommend designs based on the selected vertical

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Merge branch 'feature/site-design-revamp' into task/16503-recommend-vertical-designs

antonis committed:
Extracts the recommendation logic to a separate file

antonis committed:
Tests the recommended designs logic

antonis committed:
Adds a mechanism for randomising the order of layouts in specific categories

antonis committed:
Randomises the order of site designs except for the recommended category

antonis committed:
Also save the ordered layouts per category for efficiency

antonis committed:
Unit tests for which categories should be randomised

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Merge branch 'feature/site-design-revamp' into task/16503-recommend-vertical-designs

# Conflicts:
#	build.gradle

antonis committed:
Merge branch 'task/16503-recommend-vertical-designs' into task/16506-randomise-designs-order

antonis committed:
Unselect the layout after the choose action if fired

antonis committed:
Merge branch 'task/16503-recommend-vertical-designs' into task/16506-randomise-designs-order

antonis committed:
Adds recommended property in the enhanced_site_creation_site_design_selected event

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Merge branch 'feature/site-design-revamp' into task/16503-recommend-vertical-designs

antonis committed:
Merge branch 'task/16503-recommend-vertical-designs' into task/16506-randomise-designs-order

antonis committed:
Merge branch 'task/16503-recommend-vertical-designs' into task/16528-track-recommended-property

antonis committed:
Hides keyboard on back

ovitrif committed:
Merge pull request #16531 from wordpress-mobile/task/16506-randomise-designs-order

[Site Design Revamp] Randomize order of designs for categories

ovitrif committed:
Merge pull request #16538 from wordpress-mobile/task/16528-track-recommended-property

[Site Design Revamp] Send recommended property with Tracks event on design selection

ovitrif committed:
Merge pull request #16510 from wordpress-mobile/task/16503-recommend-vertical-designs

[Site Design Revamp] Recommends designs based on chosen vertical

antonis committed:
Moves the site design picker loadSavedState call to the correct place

ovitrif committed:
Fix screens title visibility on the new & updated site creation screens

ovitrif committed:
Merge pull request #16546 from wordpress-mobile/fix/crash-on-site-intent-rotation

Fixes crash occurring when rotating the Site Intent Question screen

antonis committed:
Updates the UI on vertical change

antonis committed:
Adds unit test verifying that the vertical change triggers a reload

ovitrif committed:
Merge pull request #16549 from wordpress-mobile/fix/16540-handle-vertical-change

[Site Design Revamp] Updates the UI on vertical change

ovitrif committed:
Align collapsing titles on site creation screens to textStart

ovitrif committed:
Align header title & subtitle on site creation screens to textStart

ovitrif committed:
Use site creation header v2 in site theme screen for consistency

mkevins committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--add-helper-text-to-bottom-as-list-item

mkevins committed:
Fix detekt issues

mkevins committed:
Fix lint

ovitrif committed:
Rename appbar title id in site theme titlebar to avoid DuplicateIncludedIds lint issue

ovitrif committed:
Remove unused dimension resource siq_title_padding_start

ovitrif committed:
Remove redundant horizontal margins override in the site creation header layout

antonis committed:
Merge pull request #16550 from wordpress-mobile/issue/16464-site-creation-tablet-screen-titles

[Site Creation] Fix screen titles visibility on tablet for the new & updated screens

antonis committed:
Merge branch 'feature/site-design-revamp' into issue/16551-align-site-creation-screen-titles-left

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt

antonis committed:
Merge pull request #16556 from wordpress-mobile/issue/16551-align-site-creation-screen-titles-left

[Site Creation] Align screen titles  to left (right for RTL)

mkevins committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--add-helper-text-to-bottom-as-list-item

mkevins committed:
Make some style adjustments

mkevins committed:
Only show footer when list is populated

This prevents an issue where the fetched categories arriving later after
the layout caused the list to be scrolled to the bottom (maintaining a
view of the footer as the new items were added).

ovitrif committed:
Merge pull request #16555 from wordpress-mobile/feature/site-design-revamp--add-helper-text-to-bottom-as-list-item

[Site design revamp] Add helper text to the bottom of the theme picker

ovitrif committed:
Replace SiteCreationActivity dependency injection with Hilt

ovitrif committed:
Inject SiteCreationVM with Hilt

ovitrif committed:
Replace dagger with hilt for SiteCreationIntents fragment & viewModel

ovitrif committed:
Replace dagger with hilt for SiteCreationSiteName fragment & viewModel

ovitrif committed:
Fix SiteCreationMainVM injection with Hilt

ovitrif committed:
Cleanup AppComponent

ovitrif committed:
Replace dagger with hilt for HomePagePicker fragment & viewModel

ovitrif committed:
Replace dagger with hilt for SiteCreationDomains fragment & viewModel

ovitrif committed:
Replace dagger with hilt for SiteCreationPreview fragment & viewModel

ovitrif committed:
Fix whitespace indentation in SiteCreationDomainsFragment

ovitrif committed:
Remove unused import in SiteCreationPreviewFragment

antonis committed:
Merge pull request #16575 from wordpress-mobile/refactor/site-creation-screens-di-with-hilt

[Site Creation] Replace Dagger with Hilt for Dependency Injection

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Add temporary feature flag to facilitate testing

In order to make testing more convenient, this adds a temporary flag
that can be toggled in the debug settings of the app. It also adds a
hard-coded behavior to clear the Glide cache (memory and disk), for the
purpose of testing the flow with a fresh start each time.

mkevins committed:
Disable volley cache for MShots loader

For some reason, deleting the Glide cache was not enough, and disabling
the volley cache was still not enough to prevent a cache hit. I suspect
that there is an additional cache later on the network side of things.
Nevertheless, we may want to keep the volley cache disabled, to prevent
3 layers of duplicate cache.

mkevins committed:
Add reminder note about nested recycler viewholder pool

The nested recycler views can only utilized a shared pool if they are
given the same instance of the pool. We can either remove the line, or
keep a reference to the instance at a higher level in the hierarchy, so
that the children recycler views can share a pool of view holders.

mkevins committed:
Implement preloader function in main site creation view model

mkevins committed:
Invoke preloading at the start, and cancel before theme picker loads

mkevins committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--preload-mshots

antonis committed:
Sets the loading skeleton to the size of the recommended row

ovitrif committed:
Merge pull request #16597 from wordpress-mobile/task/site-design-revamp-loading-skeleton

[Site Design Revamp] Sets the loading skeleton to the size of the recommended row

antonis committed:
Sets the correct height in the Page Layout Picker

mkevins committed:
Merge pull request #16599 from wordpress-mobile/fix/site-design-skeleton-regression

Fixes a UI regression by setting the correct height in the Page Layout Picker skeleton view

mkevins committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--preload-mshots

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchHomePageLayoutsUseCase.kt
#	WordPress/src/test/java/org/wordpress/android/ui/sitecreation/theme/FetchHomePageLayoutsUseCaseTest.kt

antonis committed:
Fixes Lint issue

antonis committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--preload-mshots

# Conflicts:
#	WordPress/build.gradle

antonis committed:
Updates view model test with the new parameters

mkevins committed:
Merge branch 'feature/site-design-revamp--preload-mshots' of github.com:wordpress-mobile/WordPress-Android into feature/site-design-revamp--preload-mshots

mkevins committed:
Move preload cancellation inside when expression

mkevins committed:
Set preloading null outside of network check

mkevins committed:
Revert "Add temporary feature flag to facilitate testing"

This reverts commit 2383dc9a2311771d4134780fb33e60fa018ebe54.

mkevins committed:
Revert "Disable volley cache for MShots loader"

This reverts commit aa00844433132f05d0a4efa075faff0047740dc7.

mkevins committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Merge branch 'feature/site-design-revamp' into feature/site-design-revamp--preload-mshots

mkevins committed:
Remove instantiation of nested recycledViewPools

mkevins committed:
Merge pull request #16584 from wordpress-mobile/feature/site-design-revamp--preload-mshots

[Site design revamp]  Preload MShots when the site creation activity starts

ovitrif committed:
Add divider on top of helper text of site theme screen

ovitrif committed:
Increase padding of helper text to 16dp

ovitrif committed:
Increase font size of category title on site theme

ovitrif committed:
Merge branch 'trunk' into feature/site-design-revamp

ovitrif committed:
Fix increase padding of footer to 16dp

ovitrif committed:
Remove TODO comment from init of SiteCreationMainVM that is no longer relevant

ovitrif committed:
Remove if condition for cancelling the preview preloading job

ovitrif committed:
Add bool prop for useLargeCategoryHeading to keep LayoutCategoryAdapter concerns tidy

ovitrif committed:
Move handling of row dividers to bind method block

ovitrif committed:
Suppress LongParameterList check for LayoutsItemViewHolder ctor

antonis committed:
Merge pull request #16619 from wordpress-mobile/feature/site-design-revamp--small-tweaks

[Site Creation] Tweak UI of Site Theme to match design specs

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Uses the custom Glide loader for preloading the MShots

antonis committed:
Removes unused import

ovitrif committed:
Merge pull request #16624 from wordpress-mobile/task/preload-mshot

[Site design revamp] Use the custom Glide loader for preloading the thumbnails

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

mkevins committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Adds release notes

antonis committed:
Merge pull request #16667 from wordpress-mobile/task/site-design-revamp-release-notes

Adds release notes

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis committed:
Adds margin between recommended design title and subtitle

antonis committed:
Aligns overlay and stroke visibility in design selection

twstokes committed:
Merge pull request #16692 from wordpress-mobile/task/site-design-revamp-subtitle-margin

[Site design revamp] Adds margin between recommended design title and subtitle

twstokes committed:
Merge pull request #16693 from wordpress-mobile/task/site-design-revamp-remove-overlay

[Site design revamp] Remove selection border when collapsing themes

mkevins committed:
Save view model data in home page picker fragment on orientation changes

mkevins committed:
Remove writing theme picker vm to activity bundle

antonis committed:
Merge pull request #16700 from wordpress-mobile/feature/site-design-revamp--fix-preview-orientation-change-bug

[Site Design Revamp] Fix blank theme preview after orientation change

antonis committed:
Merge branch 'trunk' into feature/site-design-revamp

antonis reviewed (approved):
After testing the latest changes according to the test plan I'm proceeding with merging this feature 🎉
